src/operations/aggregate.ts

```diff
@@ -34,14 +34,16 @@ export interface AggregateOptions extends CommandOperationOptions {
 }
 
 /** @internal */
-export class AggregateOperation<T = Document> extends CommandOperation<AggregateOptions, T> {
+export class AggregateOperation<T = Document> extends CommandOperation<T> {
+  options: AggregateOptions;
   target: string | typeof DB_AGGREGATE_COLLECTION;
   pipeline: Document[];
   hasWriteStage: boolean;
 
   constructor(parent: OperationParent, pipeline: Document[], options?: AggregateOptions) {
     super(parent, options);
 
+    this.options = options || {};
```

Comment:
Very nit: I'll refrain from picking them all out but is this better suited for `??`

Reply:
I changed this everywhere in the whole codebase, hence the dramatic increase in file count :)